configure.ac

```diff
@@ -484,7 +491,7 @@ if test "x$ac_compiler_gnu" = "xyes"; then
         if test "x$enable_debug" = "xyes"; then
                 CFLAGS_ADD="${CFLAGS_ADD} -g"
         else
-                CFLAGS_ADD="${CFLAGS_ADD} -O2 -g"
+                CFLAGS_ADD="${CFLAGS_ADD} -O2"
```

Comment:
I still believe it is a regression, because these configure options --enable-debug/--disable-debug suggest they affect debug symbol generation (by their name), which is not working anymore. Without changing `-g`, they could be renamed to --enable-optimization/--disable-optimization. 

Though I am open to the initiative, I am just yet to be convinced if it is the right path. Let's discuss irl. From implementation point of view we can easily align: if we change the default value for --enable-debug to yes: it would resolve your argument, and also the users still have a supported way for making binaries without debug symbols. 

I believe it would not be a good idea to compile with "-g" unconditionally, as the syslog-ng ecosystem is somewhat complex, it consists of many files, executables and shared libraries, so in the end it is not trivial how to do the strip properly. It is good if we give a users a tool they can use for it.

Reply:
Anything that uses automake, will strip debug symbols on `make install` (for libraries and executables), so the end result is that what is installed, will not have debug symbols, and will be almost - if not completely - indistinguishable from a build without `-g`. However, uninstalled objects will still have debug symbols, making it possible to debug an optimized build.

This is the autotools default, this is how *every single distribution* does binary packages (perhaps disabling it for things that would be way too large with `-g`, which is not the case here). I'd argue that `-g` being conditional was a bug, and the current state is the good and desired one.

Reply:
I agree, that -O2 should come with -g. Packagers expect that behavior and makes the creation of dbgsym and dbg packages easier to create.

--enable-debug is not just for symbols, it is also enabling explict, debugging related code. (e.g. msg_trace()).

Reply:
Thanks for the explanation for both of you! I did not know the make install-strip target. As none of my arguments hold, I will drop the patch from the patchset.